pkg/analytics/analytics.go

```diff
@@ -0,0 +1,139 @@
+package analytics
+
+import (
+	"bytes"
+	"crypto/md5"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/denisbrodbeck/machineid"
+)
+
+const (
+	gaUrl = "https://www.google-analytics.com/mp/collect?measurement_id=%s&api_secret=%s"
+)
+
+var (
+	kuskGAApiSecret     = "" // value needs to be passed with LDFLAG set to github.com/kubeshop/kusk-gateway/pkg/analytics.kuskGAApiSecret={ACTUAL SECRET}
```

Comment:
Is it a good idea, even in the context of a single build, hardcoding these values? I feel this would fit better as an environment variable?

Reply:
The values are passed in compile time as LD flags. Far from hard coded value. I was thinking of using LD_FLAGS for versioning and similar.

Reply:
For version that makes absolute sense but for an API key that we may want to invalidate and generate a new one, probably not.

If we change the API key, it means we need a new build. Whereas if it were an env variable, we just restart the process with the new value

Reply:
Yes but we have to control these values we cannot pass them to users.

If you have safer approach then we can go with it